src/core/mod.rs

```diff
@@ -933,6 +933,29 @@ impl JwsSigningAlgorithm<CoreJsonWebKeyType> for CoreJwsSigningAlgorithm {
     }
 }
 
+///
+/// JWK allowed algorithm "alg" field, [Section 4.4 of RFC 7517](https://www.rfc-editor.org/rfc/rfc7517#section-4.4)
+///
+/// All JSON Web Algorithms listed in [RFC 7518](https://www.rfc-editor.org/rfc/rfc7518) can be encoded by this enum.
+///
+/// There are a few caveats, which can be deduced from [the "Algorithm Usage Location(s)" part of Section 7.1.1](https://www.rfc-editor.org/rfc/rfc7518#section-7.1.1):
+///  - According to the specification, algorithms might have multiple purposes (like both JWE and JWS).
+///    No such algorithm is currently listed, but it if were to happen, the current flattened enum approach would stop working
+///  - According to the specification, algorithms which aren't used for JWE nor JWS may be added to the list.
+///    No such algorithm is currently listed.
+/// According to the "Algorithm Usage Location(s)" section of  https://www.rfc-editor.org/rfc/rfc7518#section-7.1.1
+///
+#[derive(Clone, Debug, Deserialize, Serialize, PartialEq, Eq)]
+#[serde(untagged)]
+pub enum CoreJwkAlgorithm {
```

Comment:
this type serves a similar purpose as `crate::jwt::JsonWebTokenAlgorithm`, with the addition of the key management algorithms. it doesn't seem like the key management algorithms will ever actually be included in a JWK, at least within the context of OIDC, since these are typically private keys already held by the client and used to decrypt the CEK (iiuc, please correct me if I'm wrong). I think it should be fine to use `JsonWebTokenAlgorithm` instead of introducing a new type. the `deserialize_option_or_none` suggestion above will also prevent deserialization errors in the event that algorithms not understood by this crate (including JWE key management algorithms) are used.

as far as handling algorithms that could be used by both JWE and JWS, the "proper" solution is probably to look at the `use` field (if provided) to disambiguate. but, that would require a more complex `Deserialize` implementation that isn't worth adding at this point.

Reply:
> it doesn't seem like the key management algorithms will ever actually be included in a JWK, at least within the context of OIDC, since these are typically private keys already held by the client and used to decrypt the CEK (iiuc, please correct me if I'm wrong)

I had no idea which ones made sense to include, so defaulted back to doing as the spec said, which wasn't as clear as I wish it were. 

> I think it should be fine to use JsonWebTokenAlgorithm instead of introducing a new type.

I'm happy to use the existing type, I looked for it but failed to find it

Reply:
> I'll trust you on this one: You have a lot more expertise than me on the matter, I had no idea whether key management algorithms belonged there.

I think technically any key can be formatted as a JWK, but I wouldn't expect to find a key management one in a JWK set returned by the provider's `jwks_uri`, which is the primary (sole?) purpose of the `JsonWebKey` type in this crate. in any case, this crate probably wouldn't have use for such a key (so it's fine to ignore via `VecSkipError`).